File: courses/codes/java-class/BankAccount.java

public class BankAccount {
	
	private double balance;
	private double interestRate;
	
	public BankAccount() {
		this.balance = 1000;
		this.interestRate = 0.05;
	}
	
	public BankAccount(double balance) {
		this.balance = balance;
		this.interestRate = 0.05;
	}
	
	public BankAccount(double balance, double rate) {
		this.balance = balance;
		this.interestRate = rate;
	}
	
	public double getBalance() {
		return this.balance;
	}
	
	public double getInterestRate() {
		return this.interestRate;
	}
	
	public void deposit(double amt) {
		this.balance += amt;
	}
	
	public void withdraw(double amt) {
		this.balance = Math.max(0, this.balance - amt);
	}
	
	public void withdrawAll() {
		this.balance = 0;
		// other ways of doing it
		// this.withdraw(this.balance);
		// this.withdraw(this.getBalance());
	}
	
	public void applyInterest() {
		this.balance *= (1 + this.interestRate);
	}
	
	
}

File: courses/codes/python-class/bank_account.py
class BankAccount:

    def __init__(self, balance=1000, interest_rate=0.05):
        self.balance = balance
        self.interest_rate = interest_rate

    def get_balance(self):
        return self.balance

    def get_interest_rate(self):
        return self.interest_rate

    def deposit(self, amt):
        self.balance += amt

    def withdraw(self, amt):
        self.balance = max(0, self.balance - amt)

    def withdraw_all(self):
        self.balance = 0
        # self.withdraw(self.balance)             # another way of doing it
        # self.withdraw(self.get_balance())       # another way of doing it

    def apply_interest(self):
        self.balance *= (1 + self.interest_rate)